lib/msal-browser/src/naa/mapping/NestedAppAuthAdapter.ts

```diff
@@ -0,0 +1,252 @@
+            clientCapabilities: this.clientCapabilities,
+            extraParameters: extraParams,
+        };
+
+        return tokenRequest;
+    }
+
+    public toNaaTokenRequest(
+        request: PopupRequest | RedirectRequest
+    ): TokenRequest {
+        let extraParams: Map<string, string>;
+        if (request.extraQueryParameters === undefined) {
+            extraParams = new Map<string, string>();
+        } else {
+            extraParams = new Map<string, string>(
+                Object.entries(request.extraQueryParameters)
+            );
+        }
+
+        const tokenRequest: TokenRequest = {
+            userObjectId: request.account?.homeAccountId,
+            clientId: this.clientId,
+            authority: request.authority,
+            scope: request.scopes.join(" "),
+            correlationId:
+                request.correlationId !== undefined
+                    ? request.correlationId
+                    : "",
+            prompt: request.prompt !== undefined ? request.prompt : "",
+            nonce: request.nonce !== undefined ? request.nonce : "",
+            claims: request.claims !== undefined ? request.claims : "",
+            state: request.state !== undefined ? request.state : "",
+            authenticationScheme:
+                request.authenticationScheme !== undefined
+                    ? request.authenticationScheme
+                    : "",
+            clientCapabilities: undefined,
+            extraParameters: extraParams,
+        };
+
+        return tokenRequest;
+    }
+
+    public fromNaaTokenResponse(
+        request: TokenRequest,
+        response: TokenResponse
+    ): AuthenticationResult {
+        const expiresOn = new Date(
+            TimeUtils.nowSeconds() + (response.expires_in || 0) * 1000
+        );
+
+        const account = this.fromNaaAccountInfo(response.account);
+
+        const authenticationResult: AuthenticationResult = {
+            authority: response.account.environment,
+            uniqueId: response.account.homeAccountId,
+            tenantId: response.account.tenantId,
+            scopes: response.scope.split(" "),
+            account: this.fromNaaAccountInfo(response.account),
+            idToken: response.id_token !== undefined ? response.id_token : "",
+            idTokenClaims:
+                account.idTokenClaims !== undefined
+                    ? account.idTokenClaims
+                    : {},
+            accessToken: response.access_token,
+            fromCache: true,
+            expiresOn: expiresOn,
+            tokenType:
+                request.authenticationScheme !== undefined
+                    ? request.authenticationScheme
+                    : "Bearer",
+            correlationId: request.correlationId,
+            requestId: "",
+            extExpiresOn: expiresOn,
+            state: response.state,
+        };
+
+        return authenticationResult;
+    }
+
+    /*
+     *  export type AccountInfo = {
```

Comment:
This is for quick and easy reference.   Do these get left behind after bundling/minification?  I don't remember.

Reply:
This type of comment tends to get outdated pretty quick so I prefer to avoid these but I'll leave it up to you and TAOS. Not sure what minifier TAOS is using but I would expect it to strip comments so that part is not a concern. Our minifier only runs on the UMD output which I assume you aren't using for this use case.